channeldb/channel.go

```diff
@@ -705,15 +797,15 @@ func (c *OpenChannel) putChanStatus(status ChannelStatus) error {
 			return err
 		}
 
-		channel.ChanStatus = status
+		channel.ChanStatus |= status
 
 		return putOpenChannel(chanBucket, channel)
 	}); err != nil {
 		return err
 	}
 
 	// Update the in-memory representation to keep it in sync with the DB.
-	c.ChanStatus = status
+	c.ChanStatus |= status
```

Comment:
I think this assumes that `c.ChanStatus == channel.Status` start in the same state. To make sure writes from multiple sources are seen by other `OpenChannel` objects, we should capture the ondisk value to set in memory.
```golang
var chanStatus ChannelStatus
...
        chanStatus = channel.ChannelStatus | status
        channel.ChannelStatus = chanStatus
...
c.ChanStatus = chanStatus
```
Also applies to `MarkDataLoss`

Reply:
Since the value of `ChanStatus` is now mutable, we may also need to put it behind a mutex, similar to `ShortChanID()`

Reply:
Done!

Reply:
nice! in my head i expected that diff to be much bigger, works out nicely that we never set `ChanStatus` manually in tests